.github/workflows/tag-makedist.yaml

```diff
@@ -61,12 +52,67 @@ jobs:
                       --enable-picky-compiler=yes \
                       --enable-platform-aws \
                       --with-cuda=/usr/local/cuda/
+          make distcheck V=1
+          make dist
+
+      - name: Generate source packages
+        if: matrix.generate_packages
+        run: |
+          echo "Attempting to generate packages for tag: ${{ github.ref_name }}"
+          ./contrib/scripts/generate_source_packages.sh "${{ github.ref_name }}"
+
+      - name: Prepare release notes
+        if: matrix.generate_packages
+        run: |
+          VERSION=${{ github.ref_name }}
+          awk -v version="$VERSION" '
+          BEGIN { found=0; printing=0; }
+          $0 ~ "^# " version " \\(" { found=1; printing=1; print; next }
+          printing==1 && $0 ~ "^# v[0-9]" { printing=0; exit }
+          printing==1 { print }
+          END {
+            if (!found) {
+              print "No specific release notes found for version " version > "/dev/stderr"
+              exit 1
+            }
+          }' RELEASENOTES.md > RELEASE_NOTES.md
 
-      - name: Call `make distcheck`
-        run: make distcheck V=1
+      - name: List generated files
+        if: matrix.generate_packages
+        run: ls -la *.tar* *.dsc *.rpm || true
 
-      - uses: actions/upload-artifact@v4
+      - name: Upload artifacts
+        if: matrix.generate_packages
+        uses: actions/upload-artifact@v4
         with:
-          name: "${{ github.ref_name }} make dist output"
-          path: "*.tar*"
+          name: packages-${{ matrix.name }}
+          path: |
+            *.tar*
+            *.dsc
+            *.rpm
+            RELEASE_NOTES.md
           if-no-files-found: error
+
+  create-release:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: packages-ubuntu
```

Comment:
Would it be straight-forward to have this name be set based on which OS in the matrix we set `generate_packages: true` for? Otherwise we'll need to be aware that if we decide to change the OS we generate the distribution packages on (e.g. changing to `amazonlinux`), we'll need to also change this hard-coded value.